R/class-workbook.R

```diff
@@ -1004,16 +1004,16 @@ wbWorkbook <- R6::R6Class(
 
       ## xl/comments.xml
       if (nComments > 0 | nVML > 0) {
+        sel <- vapply(wb$comments, function(x) length(x) > 0, NA)
+        comments <- self$comments[sel]
         # TODO use seq_len() or seq_along()?
-        for (i in seq_len(nSheets)) {
-          if (length(self$comments[[i]])) {
-            fn <- sprintf("comments%s.xml", i)
+        for (i in seq_along(comments)) {
+          fn <- sprintf("comments%s.xml", i)
```

Comment:
This is no longer valid. If we remove a comment 2 of three, the remaining comment will still be named 1 and 3 and worksheet_rels still has these entries. Guess we need to select worksheet_rels comment names once again.